internal/xds/translator/cluster.go

```diff
@@ -220,8 +225,17 @@ func buildXdsCluster(args *xdsClusterArgs) (*buildClusterResult, error) {
 		}
 	}
 
+	// TransportSocket is required for auto HTTP config
+	if requiresAutoHTTPConfig && len(cluster.TransportSocketMatches) > 0 && cluster.TransportSocket == nil {
```

Comment:
Should this situation even happen? If we're enabling this feature only when ALPN is enabled, doesn't that mean that we have to have a transport socket?

Reply:
IIUC, EG set transport socket only for `DynamicResolver` endpoints, but it's a must for auto HTTP config, that why we need these lines.